src/PolarQuad.cpp

```diff
@@ -0,0 +1,631 @@
+ * @details This method is a helper function to allow OpenMOC users to assign
+ *          the PolarQuad's sin thetas in Python. A user must initialize a
+ *          NumPy array of the correct size (i.e., a float64 array the length
+ *          of the number of polar angles) as input to this function. This
+ *          function then fills the NumPy array with the data values for the
+ *          PolarQuad's sin thetas. An example of how this function might be
+ *          called in Python is as follows:
+ *
+ * @code
+ *          sin_thetas= numpy.array([0.05, 0.1, 0.15, ... ])
+ *          polar_quad = openmoc.PolarQuad()
+ *          polar_quad.setNumPolarAngles(len(sin_thetas))
+ *          polar_quad.setSinThetas(sin_thetas)
+ * @endcode
+ *
+ * @param sin_thetas the array of sines of each polar angle
+ * @param num_polar the number of polar angles
+ */
+void PolarQuad::setSinThetas(FP_PRECISION* sin_thetas, int num_polar) {
+
+  if (_num_polar != num_polar)
+    log_printf(ERROR, "Unable to set %d sin thetas for PolarQuad "
+               "with %d polar angles", num_polar, _num_polar);
+
+  /* Deallocate memory if it was allocated previously */
+  if (_sin_thetas != NULL)
+    delete [] _sin_thetas;
+
+  /* Initialize memory for arrays */
+  _sin_thetas = new FP_PRECISION[_num_polar];
+
+  /* Extract sin thetas from user input */
+  for (int i=0; i < _num_polar; i++)
+    _sin_thetas[i] = FP_PRECISION(sin_thetas[i]);
+
+}
+
+
+/**
+ * @brief Set the PolarQuad's array of weights for each angle.
+ * @details This method is a helper function to allow OpenMOC users to assign
+ *          the PolarQuad's angular weights in Python. A user must initialize a
+ *          NumPy array of the correct size (i.e., a float64 array the length
+ *          of the number of polar angles) as input to this function. This
+ *          function then fills the NumPy array with the data values for the
+ *          PolarQuad's weights. An example of how this function might be
+ *          called in Python is as follows:
+ *
+ * @code
+ *          weights = numpy.array([0.05, 0.1, 0.15, ... ])
+ *          polar_quad = openmoc.PolarQuad()
+ *          polar_quad.setNumPolarAngles(len(weights))
+ *          polar_quad.setWeights(weights)
+ * @endcode
+ *
+ * @param weights the array of weights for each polar angle
+ * @param num_polar the number of polar angles
+ */
+void PolarQuad::setWeights(FP_PRECISION* weights, int num_polar) {
+
+  if (_num_polar != num_polar)
+    log_printf(ERROR, "Unable to set %d weights for PolarQuad "
+               "with %d polar angles", num_polar, _num_polar);
+
+  /* Deallocate memory if it was allocated previously */
+  if (_weights != NULL)
+    delete [] _weights;
+
+  /* Initialize memory for arrays */
+  _weights = new FP_PRECISION[_num_polar];
+
+  /* Extract weights from user input */
+  for (int i=0; i < _num_polar; i++)
+    _weights[i] = FP_PRECISION(weights[i]);
+}
```

Comment:
Might consider adding a check to see if the weights sum to 1.0. Could be useful in case users think the weights have to sum to pi.